reconcile/openshift_prometheusrules.py

```diff
@@ -0,0 +1,355 @@
+    # Generate the component-slo-namespace.libsonnet file
+    try:
+        template = template_env.get_template(LIBSONNET_TEMPLATE_FILE)
+        component_name = pp['component']
+        namespace_name = namespace['name']
+
+        http_metric_name = pp['availability'][0].get('metric')
+        availability_error_budget = pp['availability'][0].get(
+          'errorBudget'
+          )
+
+        selectors = []
+        for k, v in json.loads(
+          pp['availability'][0].get('selectors')
+          ).iteritems():
+            selectors.append("%s=\"%s\"" % (k, v))
+
+        outputText = template.render(
+          metric=http_metric_name,
+          selectors=json.dumps(selectors, ensure_ascii=True),
+          errorBudget=availability_error_budget
+          )
+        libsonnet_file_path = str(
+          temp_dir_path + "/" + component_name +
+          "-slo-" + namespace_name + ".libsonnet"
+          )
+        with open(libsonnet_file_path, 'w') as f:
+            f.writelines(outputText)
+        f.close()
+        return libsonnet_file_path
+    except Exception as e:
+        raise Exception("Template {} not found {}"
+                        .format(LIBSONNET_TEMPLATE_FILE, e)
+                        )
+
+
+def fetch_desired_state(results, ri, temp_dir_path):
+    logging.debug(['Constructing desired state'])
+    # Iterate over all available Performance Parameters
+    for pp in results['performance_parameters_v1']:
+        for namespace in pp['app']['namespaces']:
+            # Load templates from the package
+            package_templates_dir = os.path.dirname(templates.__file__)
+            template_loader = jinja2.FileSystemLoader(
+              searchpath=package_templates_dir
+              )
+            template_env = jinja2.Environment(loader=template_loader)
+
+            libsonnet_file_path = generate_libsonnet_file(
+              pp,
+              namespace,
+              temp_dir_path,
+              template_env
+              )
+
+            jsonnet_file_path = generate_jsonnet_file(
+              pp,
+              namespace,
+              libsonnet_file_path,
+              temp_dir_path,
+              template_env
+              )
+
+            # Generate the manifests
+            cmd = ['jsonnet', '-J', 'vendor/', jsonnet_file_path]
+            output = json.loads(
+              subprocess.check_output(cmd, cwd=temp_dir_path)
+              )
+
+            resource_name = output['metadata']['name']
+
+            openshift_resource = OR(
+              output,
+              QONTRACT_INTEGRATION,
+              QONTRACT_INTEGRATION_VERSION,
+              error_details=resource_name
+              )
+
+            cluster = namespace['cluster']
+            cluster_observability_namespace = cluster.get(
+              'observabilityNamespace'
+              )
+
+            ri.add_desired(
+                cluster_observability_namespace['cluster']['name'],
```

Comment:
is there a case where the observability namespace is in a different cluster?
if not, we can probably simplify a lot of things.

Reply:
>is there a case where the observability namespace is in a different cluster?

Yes, Hive. We need to keep this until we get to 4.x with one prometheus per cluster.